File: py/1261.py
m, n = map(int, input().split())

hayPoint = dict()

while m > 0:
    role, salary = input().split()
    salary = int(salary)

    hayPoint[role] = salary
    m -= 1

employee = [0 for i in range(n)]
index = 0
while index != n:
    sentence = input()

    if sentence == '.':
        index += 1

    else:
        sentence = sentence.split()
        
        for i in sentence:
            if i in hayPoint:
                employee[index] += hayPoint[i]

print(*employee, sep='\n')

File: cpp/1261.cpp
#include <iostream>
#include <string>
#include <map>

using namespace std;

int main()
{
    map<string, int> hay_point;
    int i, m, n, salary;
    cin >> m >> n;
    string role, word;

    while (m > 0) {
        cin >> role >> salary;
        hay_point[role] = salary;
        m--;
    }

    int employee[n] = {0}, index = 0;

    while (index != n) {
        cin >> word;

        if (word == ".") {
            index++;
        } else {
            if (hay_point.find(word) != hay_point.end()) {
                employee[index] += hay_point[word];
            }
        }
    }

    for (i = 0; i < n; i++) {
        cout << employee[i] << '\n';
    }

    return 0;
}